extensions/replication/ReplicationAPI.js

```diff
@@ -84,9 +87,40 @@ class ReplicationAPI {
             log.error(errorMsg, { method: 'ReplicationAPI.sendDataMoverAction' });
             return cb(new Error(errorMsg));
         }
+
+        const { reqId } = action.getContext();
+
         if (locationConfig.isCold) {
+            // Temp special case: current tape implementation does not allow
+            // zero-byte objects yet. Bypass forwarder and send directly
+            // to status topic.
+            if (contentLength === 0 && locationConfig.type === 'dmf') {
+                topic = `${coldStorageStatusTopicPrefix}${toLocation}`;
+                const message = {
+                    op: 'archive',
+                    requestId: reqId,
+                    // omit date as it makes testing difficult
```

Comment:
sinon's fake timers make this a bit easier https://sinonjs.org/releases/latest/fake-timers/